gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/test/java/io/gravitee/am/gateway/handler/manager/dictionary/I18nDictionaryManagerTest.java

```diff
@@ -0,0 +1,130 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.gateway.handler.manager.dictionary;
+
+import io.gravitee.am.common.event.I18nDictionaryEvent;
+import io.gravitee.am.gateway.handler.vertx.view.thymeleaf.GraviteeMessageResolver;
+import io.gravitee.am.model.Domain;
+import io.gravitee.am.model.I18nDictionary;
+import io.gravitee.am.model.ReferenceType;
+import io.gravitee.am.model.common.event.Payload;
+import io.gravitee.am.repository.management.api.I18nDictionaryRepository;
+import io.gravitee.common.event.Event;
+import io.reactivex.Maybe;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class I18nDictionaryManagerTest {
+
+    @Mock
+    private I18nDictionaryRepository i18nDictionaryRepository;
+
+    @Mock
+    private GraviteeMessageResolver graviteeMessageResolver;
+
+    @Mock
+    private Domain domain;
+
+    @Mock
+    private Payload payload;
+
+    @InjectMocks
+    private I18nDictionaryManager i18nDictionaryManager = new I18nDictionaryManager();
+
+    @Before
+    public void setUp() {
+        when(domain.getId()).thenReturn("domain-id");
+        when(domain.getName()).thenReturn("domain-name");
+
+        when(payload.getReferenceType()).thenReturn(ReferenceType.DOMAIN);
+        when(payload.getReferenceId()).thenReturn("domain-id");
+        when(payload.getId()).thenReturn("dictionary-id");
+    }
+
+    @Test
+    public void shouldNotDeploy_wrong_domain() {
+        Payload payload = mock(Payload.class);
+        when(payload.getReferenceType()).thenReturn(ReferenceType.DOMAIN);
+        when(payload.getReferenceId()).thenReturn("wrong-domain-id");
+
+        Event<I18nDictionaryEvent, Payload> event = mock(Event.class);
+        when(event.content()).thenReturn(payload);
+
+        i18nDictionaryManager.onEvent(event);
+
+        verify(i18nDictionaryRepository, never()).findById(anyString());
+        verify(graviteeMessageResolver, never()).updateDictionary(any());
+    }
+
+    @Test
+    public void shouldDeploy() {
+        shouldDeploy(I18nDictionaryEvent.DEPLOY);
+    }
+
+    @Test
+    public void shouldUpdate() {
```

Comment:
Informative: Test without assertion violates SonarQube S2699 (Tests should include assertions) rule (we may need to suppress the rule in that case). Possibly we could use assertDoesNotThrow if it were Junit5 [never tried though :) ].